src/rgw/rgw_file.cc

```diff
@@ -1264,8 +1264,11 @@ namespace rgw {
     }
 
     bool initial_off;
+    char* mk{nullptr};
+
     if (likely(!! get<const char*>(&offset))) {
-      initial_off = ! get<const char*>(offset);
+      mk = const_cast<char*>(get<const char*>(offset));
+      initial_off = !mk;
```

Comment:
what motivated this change? the const_cast looks unnecessary, unless `mk` is used somewhere else that i'm not seeing

Reply:
this was unrelated change that I took accidentally--but it allowed to inspect the path-based marker while debugging;  the const-cast seemed to be required by the compiler